File: 02_Lenguajes-de-programación/02_Java/05_bucles.java
/* BUCLES */

package Paquete;

import java.util.ArrayList;


public class Main {

    public static void main(String[] args) {
        bucleFor();
        bucleForMultiplicar();
        bucleWhile();
    }
    
    // 1. Bucle For
    public static void bucleFor() {
        ArrayList<String> animales = new ArrayList<String>();

        animales.add("perro");
        animales.add("gato");
        animales.add("tigre");

        for (String animal: animales) {
            System.out.println(animal);
        }
        
        System.out.println("Array completo " + animales);
    }
    
    // - Ejemplo para multiplicar un arreglo de numeros con for

    public static void multiplicar(int primero, int segundo) {
        System.out.println(primero * segundo);
    }
    
    public static void bucleForMultiplicar() {
        ArrayList<Integer> numeros = new ArrayList<Integer>();
        
        numeros.add(23);
        numeros.add(45);
        numeros.add(16);
        numeros.add(37);
        numeros.add(3);
        numeros.add(99);
        numeros.add(22);
        
        for (Integer numero: numeros) {
            multiplicar(numero, 2);
        }
    }

    
    // 2. Bucle While
    public static void bucleWhile() {
        int entero = 100;
        int emergencia = 911;

        while (entero <= emergencia) {
            System.out.println(entero);
            entero++;
        }
    }
}

File: 02_Lenguajes-de-programación/05_TypeScript/05_bucles.ts
/* Bucles */

// 1. Bucle for

// Ejemplo 1:
let animals:string[] = ['perro', 'gato', 'tigre'];
for (let animal of animals) {
    console.log(animal);
};

// Ejemplo 2:
let numero: number = 100;
function multiplicar1(primero : number, segundo: number): void {
    console.log(primero * segundo);
}

multiplicar1(numero, 3);

// Ejemplo 3:
let numeros_2: number[] = [23, 45, 16, 37, 3, 99, 20];
for (let numbers of numeros_2) {
    multiplicar1(numbers, 2);
}

// 2. Bucle while

// Ejemplo 1:
let emergencia: number = 911;
while (numero <= emergencia) {
    console.log(numero);
    numero++;
}
